File: 0812-rotate-string/0812-rotate-string.cpp
class Solution {
public:
    bool rotateString(string s, string goal) {
        if(s.size() != goal.size()) return false;
        string res = "";
        int n = s.size();
        while(n--){
            string f = "";
            for(int i = 0 ; i < s.size() ; i++){
                if(i == 0) f = s[i];
                else res = res + s[i];
            }
            res = res + f;
            if(res == goal) return true;
            s = res;
            res = "";
        }
        return false;
    }
};

File: 0812-rotate-string/0812-rotate-string.py
class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
        st = ''
        n = len(s)
        m = n
        while m != 0:
            st = s[1:n] + s[0]
            print(st)
            print()
            if st == goal:
                return True
            else:
                s = st
                print(s)
                print()
                st = ''
                m -= 1
        return False; 
        